common/achievement/get.ts

```diff
@@ -68,11 +68,11 @@ const getFurtherAchievements = async (user: User): Promise<Achievement[]> => {
         },
     });
 
-    const tieredAchievements = tieredTemplates.map((template) => {
+    const tieredAchievements = tieredTemplates.map(async (template) => {
         const dataAggr = template.conditionDataAggregations as Prisma.JsonObject;
         const maxValue = Object.keys(dataAggr)
             .map((key) => {
-                const val = dataAggr[key] as number;
+                const val = dataAggr[key]['valueToAchieve'] as number;
```

Comment:
What if the field is not defined?

Reply:
It should always return something. Otherwise the template was misconfigured. But in order to prevent an Error, due to a NaN being the value, a conditional return of 0 was added.